test/automation/src/sql/notebook.ts

```diff
@@ -362,15 +365,39 @@ export class NotebookView {
 	async searchInNotebook(expr: string): Promise<IElement> {
 		await this.waitForSetSearchValue(expr);
 		await this.code.dispatchKeybinding('enter');
-		let selector = `${NotebookView.actualResult} `;
+		let selector = NotebookView.searchResult;
 		if (expr) {
-			selector += '.message';
+			selector += ' .message';
 		}
-		return await this.code.waitForElement(selector, undefined);
+		return this.code.waitForElement(selector, undefined);
 	}
 
 	async waitForSetSearchValue(text: string): Promise<void> {
 		const textArea = `${NotebookView.inputBox} textarea`;
 		await this.code.waitForTypeInEditor(textArea, text);
 	}
+
+	/**
+	 * Helper function
+	 * @returns tree item ids from Notebooks View
+	 */
+	async getNotebookTreeItemIds(): Promise<string[]> {
+		return (await this.code.waitForElements(NotebookView.notebookTreeItem, false)).map(item => item.attributes['id']);
+	}
+
+	async pinNotebook(): Promise<void> {
+		await this.code.dispatchKeybinding('escape');
```

Comment:
Why the escape?

Reply:
The focus was on the Search Result View.

Reply:
Hm, that doesn't seem related to pinning/unpinning Notebooks though right? Typically what I would expect to see in the test is to focus on the Notebooks view before trying to pin a notebook instead of having the logic embedded in the pinNotebook function.

Reply:
...which I just saw you already did. Thanks!